environments/key4hep-common/packages.yaml

```diff
@@ -53,6 +53,8 @@ packages:
   xrootd:
     require: +krb5 cxxstd=20
 
+  fmt:
+    require: "@:10"
```

Comment:
Why? Gaudi can be built with 11

Reply:
But only for >39, right? I wanted to keep this compatible with 38 still.

Reply:
But this could be moved to the `key4hep-dev-external/spack.yaml` to only apply there.

Reply:
But this is in the common `packages.yaml` so it would affect every key4hep stack, I would like not to have this restriction which wouldn't be needed for either releases or nightlies.